src/Evolution/DiscontinuousGalerkin/SlopeLimiters/Weno.hpp

```diff
@@ -0,0 +1,373 @@
+  /// better suited for problems with strong shocks, and smaller values are
+  /// better suited to smooth problems.
+  struct NeighborWeight {
+    using type = double;
+    static type default_value() noexcept { return 0.001; }
+    static type lower_bound() noexcept { return 1e-6; }
+    static type upper_bound() noexcept { return 0.1; }
+    static constexpr OptionString help = {
+        "Linear weight for each neighbor element's solution"};
+  };
+  /// \brief Turn the limiter off
+  ///
+  /// This option exists to temporarily disable the limiter for debugging
+  /// purposes. For problems where limiting is not needed, the preferred
+  /// approach is to not compile the limiter into the executable.
+  struct DisableForDebugging {
+    using type = bool;
+    static type default_value() noexcept { return false; }
+    static constexpr OptionString help = {"Disable the limiter"};
+  };
+  using options = tmpl::list<Type, NeighborWeight, DisableForDebugging>;
+  static constexpr OptionString help = {"A WENO limiter for DG"};
+
+  Weno(WenoType weno_type, double neighbor_linear_weight,
+       bool disable_for_debugging = false) noexcept;
+
+  Weno() noexcept = default;
+  Weno(const Weno& /*rhs*/) = default;
+  Weno& operator=(const Weno& /*rhs*/) = default;
+  Weno(Weno&& /*rhs*/) noexcept = default;
+  Weno& operator=(Weno&& /*rhs*/) noexcept = default;
+  ~Weno() = default;
+
+  // NOLINTNEXTLINE(google-runtime-references)
+  void pup(PUP::er& p) noexcept;
+
+  /// \brief Data to send to neighbor elements
+  struct PackagedData {
+    Variables<tmpl::list<Tags...>> volume_data;
+    tuples::TaggedTuple<::Tags::Mean<Tags>...> means;
+    Mesh<VolumeDim> mesh;
+    std::array<double, VolumeDim> element_size =
+        make_array<VolumeDim>(std::numeric_limits<double>::signaling_NaN());
+
+    // NOLINTNEXTLINE(google-runtime-references)
+    void pup(PUP::er& p) noexcept {
+      p | volume_data;
+      p | means;
+      p | mesh;
+      p | element_size;
+    }
+  };
+
+  using package_argument_tags = tmpl::list<Tags..., ::Tags::Mesh<VolumeDim>,
+                                           ::Tags::SizeOfElement<VolumeDim>>;
+
+  /// \brief Package data for sending to neighbor elements
+  void package_data(const gsl::not_null<PackagedData*>& packaged_data,
+                    const db::item_type<Tags>&... tensors,
+                    const Mesh<VolumeDim>& mesh,
+                    const std::array<double, VolumeDim>& element_size,
+                    const OrientationMap<VolumeDim>& orientation_map) const
+      noexcept;
+
+  using limit_tags = tmpl::list<Tags...>;
+  using limit_argument_tags =
+      tmpl::list<::Tags::Element<VolumeDim>, ::Tags::Mesh<VolumeDim>,
+                 ::Tags::SizeOfElement<VolumeDim>>;
+
+  /// \brief Limit the solution on the element
+  bool operator()(
+      const gsl::not_null<std::add_pointer_t<db::item_type<Tags>>>... tensors,
```

Comment:
no need to `const` in the forward decl the `not_null`

Reply:
The compiler disagrees.

Reply:
Huh, well I don't know the rules for parameter packs but that is strange... :man_shrugging: